client/src/components/ElicitationModal.tsx

```diff
@@ -0,0 +1,181 @@
+import { useState, useEffect } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import DynamicJsonForm from "./DynamicJsonForm";
+import { JsonSchemaType, JsonValue } from "@/utils/jsonUtils";
+import { generateDefaultValue } from "@/utils/schemaUtils";
+import Ajv from "ajv";
+
+export interface ElicitationRequest {
+  id: number;
+  message: string;
+  requestedSchema: JsonSchemaType;
+  resolve: (response: ElicitationResponse) => void;
+}
+
+export interface ElicitationResponse {
+  action: "accept" | "reject" | "cancel";
```

Comment:
`reject` has been reverted to `decline`. It was slipped into the documentation at the last minute, but not the schema. When this was realized, the team decided to revert rather than make a post-release update to the schema.

```suggestion
  action: "accept" | "decline" | "cancel";
```